nef-plugin/SourceEditorExtension.swift

```diff
@@ -2,15 +2,77 @@
 
 import Foundation
 import XcodeKit
+import SourceEditorModels
+
+public protocol XCSourceEditorCommandDefinition {
```

Comment:
Why is this protocol needed? There is only a single implementation of it here, so why don't just adding these properties to the implementation? This is not abstracting over any concept that we may be interested in hiding.

Reply:
MenuEditorCommand defines the actions user can choose, and Xcode requires a dictionary `[XCSourceEditorCommandDefinitionKey : Any]` -for each action- to setup them. So using `XCSourceEditorCommandDefinition` we are splitting the Xcode responsibilities from our internal representation. Xcode should require an `[XCSourceEditorCommandDefinition]` and we provide it by `[MenuEditorCommand]`. At this point, and as the code will evolution -yes, sure- we could remove it; I do not have any strong preference to keep it or remove it, just think it defines the properties to adapt a model to Xcode UI extension.